### ISSUE TITLE
Bug: custom dinput file name

### ISSUE THREAD

**Issue by Evilserge** [x360ce/x360ce#333]
If yre using a custom dinput file name which can be specifield in options, x360ce will reset that field to default "asiloader.dll" at next start of application. Anyway, custom dinput is working, but you have to rename dinput field every time you want to make changes in configuration which is annoying and confuses some people.

**Comment by Nucleoprotein** [comment]
asiloader ??? You use official package ?

**Comment by Evilserge** [comment]
Yes, latest stable x360ce build. If you check dinput field it will be specifield as asiloader.dll by default

**Comment by Evilserge** [comment]
Its 3.2.9.81 32bit version.

**Comment by Squall-Leonhart** [comment]
Confirmed, this has been the behavior since ejocy added dinput piggybacking into the app.

**Comment by EmSuperCraft** [comment]
So, can somebody please fix it? Sometimes it changes WHILE playing a game and then everything crashes...
But thx for the Program, if it werent for xbox360ce I would have to through away my old Controllers :D

**Comment by Nucleoprotein** [comment]
What changes ? You close the x360ce App while playing the game so it cannot change ...

**Comment by EmSuperCraft** [comment]
It does it when the app is not running. For example just when I minimize the game window.

**Comment by EmSuperCraft** [comment]
Maybe I could fix it if I find the Location of the default .dll and replace it with the dinput dll?

**Comment by Nucleoprotein** [comment]
I can't happen, DLL file do not have any ini write functionality, it can only read ini files not write to it.
Also it cannot change loaded DLL, it impossible without unloading it first. 
Make sure config app added correct option to ini file.

**Comment by EmSuperCraft** [comment]
so why does my game crash, and when I look at the game options its using the wrong dll again?

**Comment by Nucleoprotein** [comment]
Crashing my be not related to this issue.

**Comment by EmSuperCraft** [comment]
it must be that option, because when I don't set it to dinput_x64.dll it crashes whilst starting up.

**Comment by Nucleoprotein** [comment]
And dinput_x64.dll is what ?

**Comment by EmSuperCraft** [comment]
the DInput File

**Comment by Squall-Leonhart** [comment]
Please do go ahead and break your OS install

**Comment by EmSuperCraft** [comment]
Why?

**Comment by Nucleoprotein** [comment]
... but what you use to as dinput_x64.dll, you need to provide such file... because if you change this and not provided such file then this option is useless

**Comment by EmSuperCraft** [comment]
I did put it into the same folder where I put the x360ce exe

**Comment by Nucleoprotein** [comment]
But what it is ? Some other mod etc. what you use ...
Also it need to be in game executable directory not x360ce.exe directory.

**Comment by EmSuperCraft** [comment]
I play nfs 2016 with xbox360ce. no mods.

**Comment by Nucleoprotein** [comment]
So why you use that dinput_x64.dll ???

**Comment by EmSuperCraft** [comment]
Because it didn't work without :D

**Comment by Nucleoprotein** [comment]
Send me that file ...

**Comment by EmSuperCraft** [comment]
https://github.com/x360ce/x360ce/raw/master/x360ce.App/Resources/dinput_x64.dll

**Comment by Nucleoprotein** [comment]
LOL, that file IS NOT dinput file is a wrapper to force load x360ce to games that do not load xinput*.dll using "static" import - -DO NOT EVER USE AS DINPUTFILE-

**Comment by EmSuperCraft** [comment]
so, why does it only work when I do? XD

**Comment by EmSuperCraft** [comment]
maybe if I try not to use any dinput file.

**Comment by Nucleoprotein** [comment]
Dunno but it should crash because you make a curricular loop of wrappers.
You can rename that file to dinput8.dll and copy it to game directory (this is the way it should be used)

**Comment by EmSuperCraft** [comment]
ok. asiloader.dll doesn't work. Crashes just like that. No dinput file also doesn't work. Both Crash in start menu. but when I set DInput to that .dll it starts and I can play it. but next time when I start the game its on asiloader again.

**Comment by EmSuperCraft** [comment]
how can I fix that? where is asiloader.dll? should I try to replace it with the other one?

**Comment by Nucleoprotein** [comment]
I think you crash is not related to this issue...

**Comment by EmSuperCraft** [comment]
well, what could it be related to?

**Comment by Nucleoprotein** [comment]
Try to disable Force Feedback.

**Comment by EmSuperCraft** [comment]
didn't work.

**Comment by EmSuperCraft** [comment]
Ok, it also doesn't work anymore when I change the DInput File.

**Comment by Evilserge** [comment]
H1, only reason to use custom dinput file - to use some side wrappers as a sweetfx or any other for really small specific game list.

Ive used that filed to load toca's dinput blocker for Dragon Dogma. (Fixed an issue with doble buttons promt)

**Comment by EmSuperCraft** [comment]
Ok, can somebody then look at this here? https://github.com/x360ce/x360ce/issues/522
thx for help :D

**Comment by EvoJocys** [comment]
This report is about the 3.x line, where x360ce was an xinput DLL copied into each game's folder. The current 4.x line works differently: it creates a virtual Xbox 360 controller that every game sees, with nothing copied into game folders. Game-specific problems from the DLL era do not carry over.

The latest release is 4.21 (September 2026): https://github.com/x360ce/x360ce/releases

Closing as part of a tracker clean-up after the 4.21 release. If it still happens on 4.21, reopen this issue with the version and what you see.
